src/test/java/org/broadinstitute/hellbender/utils/read/CigarUtilsUnitTest.java

```diff
@@ -337,4 +337,173 @@ public void testComputeCigar(String s1, String s2, String expectedCigar) throws
         final Cigar decode = TextCigarCodec.decode(expectedCigar);
         Assert.assertEquals(actualCigar, decode);
     }
+
+    /**
+     * This test is just to check that {@link #randomValidCigars()} dataProvider is actually
+     * producing only valid cigars.
+     * @param cigar the cigar to test.
+     */
+    @Test(dataProvider = "randomValidCigars")
+    public void testRadomValidCigarsDataProvider(final Cigar cigar) {
+        Assert.assertNull(cigar.isValid("annon", 0));
+    }
+
+    @Test(dataProvider = "randomValidAndFunkyCigars")
+    public void testLeftClip(final Cigar cigar) {
+        final int actual = CigarUtils.countLeftClippedBases(cigar);
```

Comment:
I looked at the function being tested and how the expected is calculated. Are they implemented the same way, i.e. the test isn't testing with hand calculated value (or value computed other ways)?